File: python/Swap the head and the tail.py
# Copilot: ignore file
# Kata: Swap the head and the tail
# Level: 7kyu
# Solved: 2025-09-20

def swap_head_tail(arr):
    mi = len(arr) // 2
    if len(arr) % 2 == 0:
        return arr[mi:] + arr[:mi]
    else:
        mn = [arr[mi]]
        arr.pop(mi)
        return arr[mi:] + mn + arr[:mi]

File: js/Swap the head and the tail.js
// Copilot: ignore file
// Kata: Swap the head and the tail
// Level: 7kyu
// Solved: 2025-09-20

function swapHeadAndTail(arr) {
    let mi = Math.floor(arr.length / 2);
    if (arr.length % 2 == 0) {
        return arr.slice(mi).concat(arr.slice(0, mi));
    } else {
        let mn = [arr[mi]];
        return arr.slice(mi + 1).concat(mn, arr.slice(0, mi));
    }
}